src/variety/pencils.js

```diff
@@ -0,0 +1,896 @@
+				this.inputBGcolor();
+			}
+		}
+		else if(this.puzzle.editmode){
+			if(this.mousestart || this.mousemove){ this.inputarrow_cell();}
+			else if(this.mouseend && this.notInputted()){ this.inputqnum_pencils();}
+		}
+	},
+
+	inputLineOrArrow : function(){
+		var pos, border;
+		pos = this.getpos(0);
+		if(this.prevPos.equals(pos)){ return;}
+		border = this.prevPos.getnb(pos);
+		var cell0 = border.sidecell[0], cell1 = border.sidecell[1];
+		
+		if(!border.isnull){
+			if(this.inputData===null) {
+				// Do not use setPencilArrow here, leave the border intact
+				if(border.isvert && (cell0.anum === border.LT || cell1.anum === border.RT)) {
+					if(cell0.anum === border.LT) {cell0.setAnum(-1);}
+					if(cell1.anum === border.RT) {cell1.setAnum(-1);}
+					this.inputData = 0;
+				}
+				if(!border.isvert && (cell0.anum === border.UP || cell1.anum === border.DN)) {
+					if(cell0.anum === border.UP) {cell0.setAnum(-1);}
+					if(cell1.anum === border.DN) {cell1.setAnum(-1);}
+					this.inputData = 0;
+				}
+			}
+
+			if(this.inputData===null){ this.inputData=(border.isLine()?0:1);}
+			if(this.inputData===1 && !border.ques){
+				if(border.qans === 1) {
+					var dir = this.getDrawArrowDirection(border);
+					if(dir > 0) {
+						cell1.setPencilArrow(border.isvert ? border.RT : border.DN, false);
+					} else {
+						cell0.setPencilArrow(border.isvert ? border.LT : border.UP, false);
+					}
+				} else {
+					border.setLine();
+				}
+			}
+			else if(this.inputData===0){ border.removeLine();}
+			border.draw();
+		}
+		this.prevPos = pos;
+	},
+
+	getDrawArrowDirection: function(border) {
+		if(border.sidecell[0].isnull) {return +1;}
+		if(border.sidecell[1].isnull) {return -1;}
+
+		if(border.sidecell[0].qnum > 0) {return +1;}
+		if(border.sidecell[1].qnum > 0) {return -1;}
+
+		if(border.sidecell[1].lcnt > 0 && border.sidecell[0].lcnt === 0) {return +1;}
+		if(border.sidecell[0].lcnt > 0 && border.sidecell[1].lcnt === 0) {return -1;}
+
+		if(border.sidecell[0].room.pencil) {return +1;}
+		if(border.sidecell[1].room.pencil) {return -1;}
+
+		var pos = this.getpos(0);
+		var dir = this.prevPos.getdir(pos,2);
+
+		return (dir === border.RT || dir === border.DN) ? +1 : -1;
+	},
+
+	mouseinput_clear : function(){
+		var cell = this.getcell();
+		var dir = cell.qdir || cell.anum;
+
+		if(dir>= 1 && dir <= 4) {
+			cell.setPencilArrow(0, true);
+		} else {
+			this.inputclean_cell();
+		}
+	},
+
+	inputarrow_cell_main : function(cell, dir){
+		cell.setPencilArrow(dir, this.puzzle.editmode);
+	},
+
+	inputqnum_pencils : function(){
+		var cell = this.getcell();
+		if(cell.isnull){ return;}
+
+		if(cell!==this.cursor.getc() && this.inputMode==='auto'){
+			this.setcursor(cell);
+		}
+		else{
+			this.inputcell_pencils(cell);
+		}
+	},
+	
+	inputcell_pencils : function(cell){
+		var dir = cell.qdir, num = cell.qnum, val;
+		// -4to-1:Arrow 0:? 1:何もなし 2以上:数字
+		if     (num===-2){ val = 0;}
+		else if(dir=== 5){ val = (num!==-1 ? num+1 : 2);}
```

Comment:
What is qdir value 5 about? I see it places question marks; are those meant to indicate "there's a tip here but we don't know which side"? Is that something that happens in puzzles?

Reply:
This is a holdover from Sashigane, and it means that a cell has a number, so the user can't add a pencil tip in this cell. I'm fine with removing this entirely.